app/src/main/res/layout/fragment_login.xml

```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
Why do you have so many wrapping layouts, should have 1 parent layout, (constraint) which has the input fields as direct children.

Reply:
I think the wrappers are fine. It makes it easier to position as well as add in modularity.